--- a/lib/gt_bridge/serializer.ex	2026-07-21 02:14:52.195812407 +0800
+++ b/lib/gt_bridge/serializer.ex	2026-07-21 02:14:52.206407736 +0800
@@ -29,6 +29,13 @@
     if String.valid?(data), do: data, else: ["__base64__", Base.encode64(data)]
   end
 
+  defp sanitize(%mod{} = data) do
+    data
+    |> Map.from_struct()
+    |> Map.new(fn {k, v} -> {k, sanitize(v)} end)
+    |> Map.put(:__struct__, mod)
+  end
+
   defp sanitize(data) when is_map(data) do
     Map.new(data, fn {k, v} -> {k, sanitize(v)} end)
   end
@@ -60,6 +67,13 @@
   defp desanitize(["__pid__" | rest]), do: :erlang.list_to_pid(rest)
   defp desanitize(["__base64__", encoded]), do: Base.decode64!(encoded)
 
+  defp desanitize(%mod{} = data) do
+    data
+    |> Map.from_struct()
+    |> Map.new(fn {k, v} -> {k, desanitize(v)} end)
+    |> Map.put(:__struct__, mod)
+  end
+
   defp desanitize(data) when is_map(data) do
     Map.new(data, fn {k, v} -> {k, desanitize(v)} end)
   end
